commit 5270f30130c033de15d1104d94c441877bd72496 (refs/changes/67/4167/1)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2018-06-08T11:41:15+00:00 (1 year, 4 months ago)
    
    TRITON-478 agent builds fail as non-root when postinstall.sh writes to /

diff --git a/npm/postinstall.sh b/npm/postinstall.sh
index fe6116b..a961a0f 100755
--- a/npm/postinstall.sh
+++ b/npm/postinstall.sh
@@ -6,9 +6,14 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
+if [[ -z "$npm_config_smfdir" || ! -d "$npm_config_smfdir" ]]; then
+    echo "Skipping config-agent postinstall (assuming build-time install)"
+    exit 0
+fi
+
 if [[ -n "$TRACE" ]]; then
     export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
     set -o xtrace
